### ISSUE TITLE
SPNEGO OID of MechToken is not of type KRB5

### ISSUE THREAD

**Issue by AndrewWPhillips** [jcmturner/gokrb5#451]
I am getting the above error when using Integrated Windows Authentication.  I think this is a bug. The problem appears to be at spnego.go line78.
```
		oid = t.NegTokenInit.MechTypes[0]
...
	if !(oid.Equal(gssapi.OIDKRB5.OID()) || oid.Equal(gssapi.OIDMSLegacyKRB5.OID())) {
```

The MechTypes slice has 4 entries, but entry 0 is of the wrong type.  t.NegTokenInit.MechTypes[1] == OIDKRB5 and t.NegTokenInit.MechTypes[2] == OIDMSLegacyKRB5.  I am not sure but I think that all entries should be checked not just the first.

**Comment by tooptoop4** [comment]
did u solve?

**Comment by tomqwpl** [comment]
Yes, same here. I think the list needs to be searched for the relevant OID
